junit-test/shared/src/test/scala/org/scalajs/junit/utils/JUnitTest.scala

```diff
@@ -0,0 +1,236 @@
+      buff.append(blue("Test run finished: "))
+      val failedColor = if (expectedFail == 0) blue _ else red _
+      buff.append(failedColor(s"$expectedFail failed"))
+      buff.append(blue(", "))
+      val ignoredColor = if (expectedIgnored == 0) blue _ else yellow _
+      buff.append(ignoredColor(s"$expectedIgnored ignored"))
+      buff.append(blue(s", $expectedTotal total, ${TIME_TAG}s"))
+      infoIfOrElseDebug(verbose, buff.result())
+    }
+
+    def skippedEvent: Output = Event("Skipped")
+    def successEvent: Output = Event("Success")
+    def failureEvent: Output = Event("Failure")
+
+    def infoIfOrElseDebug(p: Boolean, msg: String): Output =
+      if (p) Info(msg)
+      else Debug(msg)
+
+    override def toString: String = frameworkArgs.mkString("[", ", ", "]")
+
+    private def red(str: String): String =
+      if (noColor) str
+      else RED + str + NORMAL
+
+    private def yellow(str: String): String =
+      if (noColor) str
+      else YELLOW + str + NORMAL
+
+    private def blue(str: String): String =
+      if (noColor) str
+      else BLUE + str + NORMAL
+
+    private def magenta(str: String): String =
+      if (noColor) str
+      else MAGENTA + str + NORMAL
+
+    private def cyan(str: String): String =
+      if (noColor) str
+      else CYAN + str + NORMAL
+
+  }
+
+  protected class JunitTestRecorder(context: OutputContext)
+      extends Logger with EventHandler {
+    val buff = List.newBuilder[Output]
+
+    def ansiCodesSupported(): Boolean = true
+
+    def info(msg: String): Unit = buff += Info(encodeTimes(msg))
+
+    def warn(msg: String): Unit = buff += Warn(encodeTimes(msg))
+
+    def debug(msg: String): Unit = buff += Debug(encodeTimes(msg))
+
+    def error(msg: String): Unit = {
+      if (msg.startsWith("    at ") || msg.startsWith("    ...")) {
+        println(GREY + "Ignoring stacktrace output: " + msg + NORMAL)
+      } else {
+        buff += Error(encodeTimes(msg))
+      }
+    }
+
+    def trace(t: Throwable): Unit = buff += Trace(t.getMessage)
+
+    def handle(event: sbt.testing.Event): Unit = {
+      buff += Event(event.status().toString)
+    }
+
+    def recordDone(msg: String): Unit = {
+      buff += Done(msg)
+      checkOutput(buff.result())
+    }
+
+    private def encodeTimes(msg: String): String = {
+      val enc1 =
+        msg.replaceFirst(", took \\d+(\\.\\d*)? sec", s", took $TIME_TAG sec")
+      if (!enc1.contains("Test run finished")) enc1
+      else enc1.replaceFirst("\\d+(\\.\\d*)?s", TIME_TAG + "s")
+    }
+
+    private def checkOutput(actual: List[Output]): Unit = {
+      val expected = expectedOutput(context)
+      type ListTuple = (List[Output], List[Output])
+      def minimizeDiff(lists: ListTuple): ListTuple = {
+        lists match {
+          case (x::xs, y::ys) if x == y => minimizeDiff(xs, ys)
+          case _                        => lists
+        }
+      }
+
+      val (ex, ac) = minimizeDiff((expected, actual))
```

Comment:
Yes, but then the diff becomes a useless because it is impossible to know where it should be in the list. I solved this problem by printing in grey all the elements that were dropped by `minimizeDiff`. 

The output will be like:
![screen shot 2016-05-20 at 11 56 44](https://cloud.githubusercontent.com/assets/3648029/15424286/0d07eaf0-1e82-11e6-8d3e-d0a57e94cd1a.png)

Where the diff is clearly the additional `Event(Failure)`. I'll also remove the color from the non diff lines.